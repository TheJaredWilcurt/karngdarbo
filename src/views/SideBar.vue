<template>
  <div :class="sidebarCollapsed ? 'collapse' : ''" class="sidebar">
    <side-bar-controls />
    <repos-list />
    <repos-list-controls />
  </div>
</template>

<script>
import { mapState } from 'vuex';

import SideBarControls from '@/components/sidebar/SideBarControls.vue';
import ReposList from '@/components/sidebar/ReposList.vue';
import ReposListControls from '@/components/sidebar/ReposListControls.vue';

export default {
  name: 'SideBar',
  components: {
    'side-bar-controls': SideBarControls,
    'repos-list': ReposList,
    'repos-list-controls': ReposListControls
  },
  computed: {
    ...mapState([
      'sidebarCollapsed'
    ])
  }
};
</script>
