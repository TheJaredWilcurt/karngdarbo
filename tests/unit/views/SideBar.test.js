import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex, { Store } from 'vuex';

import SideBar from '@/views/SideBar.vue';

describe('SideBar.vue', () => {
  let store;
  let localVue;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);

    store = new Store({
      state: {
        sidebarCollapsed: false
      },
      getters: {},
      mutations: {},
      actions: {}
    });
  });

  describe('Created', () => {
    test('Default snapshot', () => {
      const wrapper = shallowMount(SideBar, {
        store,
        localVue
      });

      expect(wrapper.html())
        .toMatchSnapshot();
    });
  });
});
