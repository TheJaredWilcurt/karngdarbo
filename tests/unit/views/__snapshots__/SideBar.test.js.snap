// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar.vue Created Default snapshot 1`] = `
<div class="sidebar">
  <side-bar-controls-stub></side-bar-controls-stub>
  <repos-list-stub></repos-list-stub>
  <repos-list-controls-stub></repos-list-controls-stub>
</div>
`;
